### PR TITLE
linuxPackages_custom: fix missing argument and add test

### PR DESCRIPTION
**Pull request by roberth**
See also backport to 18.03, #38035 

The required argument 'hostPlatform' was missing from linuxPackages_custom's
call to linuxManualConfig.

In order to prevent this in the future, this commit adds
linuxPackages_custom_tinyconfig_kernel so linuxPackages_custom gets tested.

This also adds linuxConfig, to derivate default linux configurations
via make defconfig, make tinyconfig, etc.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by roberth** [comment]
@grahamcofborg build linuxPackages_custom_tinyconfig_kernel

**Comment by roberth** [comment]
@thoughtpolice ready for merge?

File: pkgs/os-specific/linux/kernel/manual-config.nix [hunk 1]
```diff
@@ -35,6 +35,8 @@ in {
   extraMeta ? {},
   # Whether to utilize the controversial import-from-derivation feature to parse the config
   allowImportFromDerivation ? false,
+  # ignored
+  features ? null,
```
Comment (thoughtpolice) [inline pkgs/os-specific/linux/kernel/manual-config.nix]: Excuse me if I'm missing something obvious -- is this change necessary? Am I missing something?

**Review by thoughtpolice — APPROVED** [review]
LGTM otherwise, thanks for adding the test, too! One small question but otherwise I think this is fine.

File: pkgs/os-specific/linux/kernel/manual-config.nix (hunk 1 above)
Comment (roberth) [reply pkgs/os-specific/linux/kernel/manual-config.nix]: `features` has become a mandatory argument for kernels. I don't remember where it is passed as an argument, but without it evaluation fails because some other code expects to be able to pass a `features` argument. I'm not sure where that is, as I said, but this [override](https://github.com/roberth/nixpkgs/blob/b5d54b04b5e2fe3463f14c09226a351f7df9e6d4/pkgs/top-level/all-packages.nix#L13618) supports the idea.
This ignores the `features` parameter entirely, which kind of seems appropriate for a custom kernel. Of course someone could write the code to validate the custom configuration against it and expose it via `passthru` but that's not going to be part of this bugfix.

Comment (thoughtpolice) [reply pkgs/os-specific/linux/kernel/manual-config.nix]: Ah, that seems OK, thanks for the follow up! (I reviewed it a bit 'out of context' since I was busy earlier and hadn't looked deeper). This otherwise looks good to me; I'll merge this (and the backport) soon...

**Comment by thoughtpolice** [comment]
*Sigh*. I merged this manually and added a `Closes` tag (7f25b26511aef36fdad4fb3999af8ce7c2dfea72), but GitHub didn't close this! Argh...

Thanks again @roberth, sorry about the delay.
